src/ERC20Whitelist.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import {IOracleWhitelist} from "./interfaces/IOracleWhitelist.sol";
+import {IApproveAndCallReceiver} from "./interfaces/IApproveAndCallReceiver.sol";
+import {Initializable} from "./base/Initializable.sol";
+import {IERC20Whitelist} from "./interfaces/IERC20Whitelist.sol";
+import {ERC20, ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Extended token contract with whitelist contract interactions
+ * @notice During whitelist period, `_beforeTokenTransfer` function will call `checkWhitelist` function of whitelist contract
+ * @notice If whitelist period is ended, owner will set whitelist contract address back to address(0) and tokens will be transferred freely
+ */
+contract ERC20Whitelist is IERC20Whitelist, ERC20Burnable, Ownable, Initializable  {
```

Comment:
We should add `ERC20Permit` too. Probably Burnable and Permit should the be default

![Screenshot 2024-07-25 at 09 00 37](https://github.com/user-attachments/assets/39188cd8-8d2b-4936-9a22-0e6d4e98e1ed)